File: swagger/models/client_purchase_record.py
from .base_model import BaseModel

from .sale import Sale

class ClientPurchaseRecord(BaseModel):
	"""
	A record of a specific client purchase

	Attributes:

		sale
		Contains details about the sale and payment for a purchase event.

		description
		The item name and description.

		account_payment
		If `true`, the item was a payment credited to an account.

		price
		The item's price before taxes and discounts.

		amount_paid
		The amount paid for the item.

		discount
		The discount amount that was applied to the item.

		tax
		The amount of tax that was applied to the item.

		returned
		The return status of the item. If `true`, this item was returned.

		quantity
		The quantity of the item purchased.
	"""

	input_map = {
		'Sale': ('sale', Sale),
		'Description': 'description',
		'AccountPayment': 'account_payment',
		'Price': 'price',
		'AmountPaid': 'amount_paid',
		'Discount': 'discount',
		'Tax': 'tax',
		'Returned': 'returned',
		'Quantity': 'quantity',
		}

	output_map = {
		'sale': 'Sale',
		'description': 'Description',
		'account_payment': 'AccountPayment',
		'price': 'Price',
		'amount_paid': 'AmountPaid',
		'discount': 'Discount',
		'tax': 'Tax',
		'returned': 'Returned',
		'quantity': 'Quantity',
		}

File: swagger/Models/ClientPurchaseRecord.php
<?php

namespace App\MindBody\Models;

/**
 * A record of a specific client purchase
 */
class ClientPurchaseRecord extends BaseModel
{
	/**
	 * Contains details about the sale and payment for a purchase event.
	 * 
	 * @var Sale
	 */
	public $sale;
	/**
	 * The item name and description.
	 * 
	 * @var string
	 */
	public $description;
	/**
	 * If `true`, the item was a payment credited to an account.
	 * 
	 * @var boolean
	 */
	public $accountPayment;
	/**
	 * The item's price before taxes and discounts.
	 * 
	 * @var number
	 */
	public $price;
	/**
	 * The amount paid for the item.
	 * 
	 * @var number
	 */
	public $amountPaid;
	/**
	 * The discount amount that was applied to the item.
	 * 
	 * @var number
	 */
	public $discount;
	/**
	 * The amount of tax that was applied to the item.
	 * 
	 * @var number
	 */
	public $tax;
	/**
	 * The return status of the item. If `true`, this item was returned.
	 * 
	 * @var boolean
	 */
	public $returned;
	/**
	 * The quantity of the item purchased.
	 * 
	 * @var integer
	 */
	public $quantity;

	protected function getInputMap()
	{
		return [
			'Sale' => ['sale', Sale::class],
			'Description' => 'description',
			'AccountPayment' => 'accountPayment',
			'Price' => 'price',
			'AmountPaid' => 'amountPaid',
			'Discount' => 'discount',
			'Tax' => 'tax',
			'Returned' => 'returned',
			'Quantity' => 'quantity',
			];
	}

	protected function getOutputMap()
	{
		return [
			'sale' => 'Sale',
			'description' => 'Description',
			'accountPayment' => 'AccountPayment',
			'price' => 'Price',
			'amountPaid' => 'AmountPaid',
			'discount' => 'Discount',
			'tax' => 'Tax',
			'returned' => 'Returned',
			'quantity' => 'Quantity',
			];
	}
}
